### PR TITLE
Sanjam/76898/TS migration of Mobile Drawer Component

### PR DESCRIPTION
**Pull request by sanjam-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   TS MIgration Of Mobile Drawer Component

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/components/src/components/mobile-drawer/mobile-drawer-body.tsx [hunk 1]
```diff
@@ -1,14 +1,12 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 
-const Body = ({ children, className }) => (
+type TBody = {
+    className: string;
```
Comment (likhith-deriv) [inline packages/components/src/components/mobile-drawer/mobile-drawer-body.tsx:5]: className is an optional prop

File: packages/components/src/components/mobile-drawer/mobile-drawer-footer.tsx [hunk 2]
```diff
@@ -2,7 +2,11 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import React from 'react';
 
-const Footer = ({ children, className }) => (
+type TFooter = {
+    className: string;
```
Comment (likhith-deriv) [inline packages/components/src/components/mobile-drawer/mobile-drawer-footer.tsx:6]: Please make all css class or styles values passed as optional

File: packages/components/src/components/mobile-drawer/mobile-drawer-submenu-section.tsx [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
-import PropTypes from 'prop-types';
 import React from 'react';
 import Text from '../text/text';
 import Icon from '../icon/icon';
 
-const SubMenuSection = props => {
-    const { submenu_toggle_class, section_title, section_icon, children } = props;
+type TSubMenuSection = React.PropsWithChildren<{
+    submenu_toggle_class: string;
+    section_title: string | React.ReactElement;
```
Comment (hubert-deriv) [inline packages/components/src/components/mobile-drawer/mobile-drawer-submenu-section.tsx:8]: `section_title: React.ReactNode`

Perhaps try this and see if it works, ReactNode should also support text input.

File: packages/components/src/components/mobile-drawer/mobile-drawer-submenu.tsx [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
 import { CSSTransition } from 'react-transition-group';
 import Text from '../text/text';
 import Icon from '../icon/icon';
 
+type TMobileDrawerSubmenu = {
+    has_subheader: boolean;
+    onToggle: (params: boolean) => void;
+    submenu_toggle_class: string;
+    submenu_icon: string;
+    submenu_title: string | React.ReactElement;
```
Comment (hubert-deriv) [inline packages/components/src/components/mobile-drawer/mobile-drawer-submenu.tsx:12]: Same here, try `submenu_title: React.ReactNode`

**Review by hubert-deriv — CHANGES_REQUESTED** [review]

File: packages/components/src/components/mobile-drawer/mobile-drawer.tsx [hunk 5]
```diff
@@ -1,16 +1,29 @@
...
+import Footer from './mobile-drawer-footer';
+import SubHeader from './mobile-drawer-subheader';
+import Item from './mobile-drawer-item';
+import SubMenu from './mobile-drawer-submenu';
+import SubMenuSection from './mobile-drawer-submenu-section';
 import Text from '../text/text';
 import Icon from '../icon/icon';
 
+type TMobileDrawer = {
+    className: string;
```
Comment (yauheni-kryzhyk-deriv) [inline packages/components/src/components/mobile-drawer/mobile-drawer.tsx:16]: Please check here and for other components what props can we mark as optional

**Review by iman-fs — APPROVED** [review]

File: packages/components/src/components/mobile-drawer/mobile-drawer.tsx [hunk 6]
```diff
@@ -1,16 +1,29 @@
...
+    id: string;
+    height: string;
+    width: string;
+    alignment: 'left' | 'right';
+    is_open: boolean;
+    title: string | boolean;
```
Comment (likhith-deriv) [inline packages/components/src/components/mobile-drawer/mobile-drawer.tsx:22]: Please recheck the type for this

File: packages/components/src/components/mobile-drawer/mobile-drawer.tsx [hunk 7]
```diff
@@ -1,16 +1,29 @@
...
+import SubHeader from './mobile-drawer-subheader';
+import Item from './mobile-drawer-item';
+import SubMenu from './mobile-drawer-submenu';
+import SubMenuSection from './mobile-drawer-submenu-section';
 import Text from '../text/text';
 import Icon from '../icon/icon';
 
+type TMobileDrawer = {
+    className: string;
+    id: string;
```
Comment (likhith-deriv) [inline packages/components/src/components/mobile-drawer/mobile-drawer.tsx:17]: id is also optional

**Review by likhith-deriv — CHANGES_REQUESTED** [review]
Please resolve the comments

**Review by likhith-deriv — APPROVED** [review]

**Pull request merged**